src/sentry/snuba/metrics/fields/base.py

```diff
@@ -247,14 +381,19 @@ def _raise_entity_validation_exception(self, func_name: str):
 
 
 class SingularEntityDerivedMetric(DerivedMetric):
-    def __init__(self, *args: Any, **kwargs: Any) -> None:
-        super().__init__(*args, **kwargs)  # type: ignore
-        self.result_type = "numeric"
-
-        if self.snql is None:
-            raise DerivedMetricParseException(
-                "SnQL cannot be None for instances of SingularEntityDerivedMetric"
-            )
+    # Pretend for the typechecker that __init__ is not overridden, such that
```

Comment:
With this patch the mis-typing of `snql` is exposed. Without it the IDE doesn't error, because `args/kwargs` is weakly typed.